tests/b2share_unit_tests/schemas/test_schemas_rest.py

```diff
@@ -68,3 +71,161 @@ def test_valid_get(app, test_communities):
+            "name":"abcd"
+        }
+        all_versions = block_schema.versions
+        with app.test_client() as client:
+            headers = [('Content-Type', 'application/json'),
+                       ('Accept', 'application/json')]
+            res = client.put(
+                url_for(
+                    'b2share_schemas.block_schema_versions_item',
+                    schema_id=str(block_schema.id),
+                    schema_version_nb=len(all_versions) - 1,
+                    _method='PUT'
+                ),
+                data=json.dumps({
+                    'json_schema': json_schema
+                }),
+                headers=headers)
+            assert res.status_code == 409
+
+
+def test_create_invalid_schema_version(app, test_communities):
+    """Test creating an invalid version of the schema."""
+    with app.app_context():
+        community_id = '2b884138-898f-4651-bf82-34dea0e0e83f'
+        json_schema = {"$schema": "http://json-schema.org/draft-04/schema#"}
+        block_schema = BlockSchema.create_block_schema(community_id, 'abc')
+        with app.test_client() as client:
+            headers = [('Content-Type', 'application/json'),
+                       ('Accept', 'application/json')]
+            res = client.put(
+                url_for(
+                    'b2share_schemas.block_schema_versions_item',
+                    schema_id=str(block_schema.id),
+                    schema_version_nb=5,
+                    _method='PUT'
+                ),
+                data=json.dumps({
+                    'json_schema': json_schema
+                }),
+                headers=headers)
+            assert res.status_code == 400
+
+
+def test_get_schemas(app, test_communities):
+    """"Test getting list of given community's schemas."""
+    with app.app_context():
+        community = Community.create_community('name1', 'desc1')
+        community_id = community.id
+        community_2 = Community.create_community('name2', 'desc2')
+        community_id2 = community_2.id
+        block_schema_1 = BlockSchema.create_block_schema(community_id, 'abc')
+        block_schema_2 = BlockSchema.create_block_schema(community_id2, 'abc2')
+        block_schema_3 = BlockSchema.create_block_schema(community_id2, 'abc3')
+        with app.test_client() as client:
+            headers = [('Content-Type', 'application/json'),
+                       ('Accept', 'application/json')]
+            res = client.get(
+                url_for(
+                    'b2share_schemas.block_schema_list',
+                    community_id=community_id2
+                ),
+                headers=headers)
+            assert res.status_code == 200
+            response_data = json.loads(res.get_data(as_text=True))
+            assert isinstance(response_data['schemas'], list)
+            assert len(response_data['schemas']) == 2
+            assert response_data['schemas'][0]['name'] == block_schema_2.name
```

Comment:
I don't see any sorting in `BlockSchema. get_all_block_schemas`. This test might fail if the schemas are sent in another order. The simplest solution would be to have a default sorting in `BlockSchema. get_all_block_schemas` based on the date of creation.